client-rpc/src/server.rs

```diff
@@ -49,6 +50,13 @@ impl Server {
     pub(crate) fn new(options: Options) -> Result<Server> {
         let network_id =
             hex::decode(&options.network_id).context(ErrorKind::SerializationError)?[0];
+        let network_type = options.network_type;
+        match &network_type[..4] {
+            "main" => init_network(Network::Mainnet),
```

Comment:
`init_network` isn't public anymore.

What about:

1. make `options.network_id` optional argument
2.  use the constants from chain_core and `init_chain_id`: 
- `"main" => init_chain_id(MAINNET_CHAIN_ID)`
- `"test" => init_chain_id(TESTNET_CHAIN_ID)`
- for devnet, require `options.network_id` and do `init_chain_id(format!("devnet-{}", options.network_id))`
3. for the RPC server, use `get_network_id()` from chain-core (after it's initialised)

What do you think?

Reply:
not bad
so it's like this

--network-type main 
or
--network-type test
or  automaticall dev mode, if network type is not specified
--network-id=ab